File: androidwork/Ex0720/app/src/main/java/com/example/ex0720/MainActivity.java
package com.example.ex0720;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState); // Activity 구동 준비!
        setContentView(R.layout.activity_main); //Java파일과 xml파일을 연결!
        // 리소스 폴더 아래의 layout폴더의 activity_main이라는 레이아웃 화면에 그려라

        // findViewById할 때 setContentView 된 xml안에서 찾음!!
        //1.만약에 main.xml에도 btn1이 있고! kakao.xml에도 btn1이 있으며 뭘 찾아옴?? 현재 setContentView된 곳
        // 2. setContentView가 없는 상황에서 findViewById하면? 에러가 남
    }
}

File: androidwork/lotto/app/src/main/java/com/example/lotto/MainActivity.kt
package com.example.lotto

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //6개의 텍스트뷰를 같은 어레이리스트를 만듬
        val lotteryNumbers = arrayListOf(number1, number2, number3, number4, number5, number6)

        val countDownTimer = object:CountDownTimer(3000,100){
            //3초짜리 타이머를 지정하고 100밀리 센컨드마다 작업하겠다.

            override fun onFinish() { //타이머가 다했을때 호출
            }
            override fun onTick(p0: Long) {
                //100밀리 세컨드마다 호출되면서 전체3초중에 얼마나 남았는지가 파라미터로 전달
                lotteryNumbers.forEach {
                    val randomNumber = (Math.random() * 45 + 1).toInt() //Integer형태로 변환
                    it.text = "$randomNumber"
                }
            }
        }

        lotteryButton.setOnClickListener {
            if (lotteryButton.isAnimating){//클릭했을때 동작하고 있는지 아닌지 알기위해서
                //동작하는 중이라면
                lotteryButton.cancelAnimation() //애니메이션 멈춤
                countDownTimer.cancel() //타이머 취소

            }else{//동작하고 있는 중이 아니라면
                lotteryButton.playAnimation()//클릭했을때 애니메이션이 실행이됨
                countDownTimer.start()

            }

        }
    }
}